### PR TITLE
java-backend bugfix: #getKLabelString

### PR DESCRIPTION
**Pull request by denis-bogdanas**
1. #getKLabelString properly handles variable arguments.
2. Added check that requires clause of a spec rule is not false.

Tested in verified-smart-contracts/k-test.

DO NOT MERGE

### CODE REVIEW

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java [hunk 1]
```diff
@@ -351,6 +351,12 @@ public ProcessProofRules invoke(GlobalContext rewritingContext, long initCounter
                 termContext.setTopConstraint(constraint);
                 //simplify the constraint in its own context, to force full evaluation of terms.
                 constraint = constraint.simplify(termContext);
+                if (constraint.isFalseExtended()) {
+                    StringBuilder sb = new StringBuilder();
+                    sb.append("Rule requires clause evaluates to false:\n");
+                    RuleSourceUtil.appendRuleAndSource(rule, sb);
+                    throw KEMException.criticalError(sb.toString());
+                }
```
Comment (daejunpark) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java:363]: This is orthogonal to the `#getKLabelString` fix, right?

File: java-backend/src/main/java/org/kframework/backend/java/builtins/MetaK.java [hunk 2]
```diff
@@ -201,6 +202,8 @@ public static KItem getKLabel(KItem kItem, TermContext context) {
     public static StringToken getKLabelString(Term term, TermContext context) {
         if (term instanceof KItem) {
             return StringToken.of(((KItem) term).kLabel().toString());
+        } else if (term instanceof KVariable){
+            return null;
```
Comment (daejunpark) [inline java-backend/src/main/java/org/kframework/backend/java/builtins/MetaK.java:206]: Could you explain in which case we need this?  An example?

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java (hunk 1 above)
Comment (denis-bogdanas) [reply java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java:363]: It is made possible by `#getKLabelString` and is also a safety measure to avoid this sort of issue in future with builtin functions.

File: java-backend/src/main/java/org/kframework/backend/java/builtins/MetaK.java (hunk 2 above)
Comment (denis-bogdanas) [reply java-backend/src/main/java/org/kframework/backend/java/builtins/MetaK.java:206]: https://github.com/runtimeverification/verified-smart-contracts/blob/b9aa9543670a4b4bcf3d1450d44f66d028404cec/k-test/k-test-spec.ini#L59-L66

**Comment by ehildenb** [comment]
Can one of the admins verify this patch?

**Comment by denis-bogdanas** [comment]
@daejunpark  Can you approve pls? We are already using new `#getKLabelString` semantics in K-gnosis. All tests on Jenkins pass.

**Review by daejunpark — APPROVED** [review]
